go/adbc/driver/flightsql/flightsql_connection.go

```diff
@@ -57,10 +58,90 @@ type connectionImpl struct {
 	supportInfo support
 }
 
+func stringMetadataHandler(xdbc_key string, flight_key string) func(arrow.Field, array.Builder) {
+	return func(column arrow.Field, builder array.Builder) {
+		if column.HasMetadata() {
+			md := column.Metadata
+			b := builder.(*array.StringBuilder)
+			if typeName, ok := md.GetValue(flight_key); ok {
+				b.Append(typeName)
+			} else if typeName, ok = md.GetValue(xdbc_key); ok {
+				b.Append(typeName)
+			} else {
+				b.AppendNull()
+			}
+		} else {
+			builder.AppendNull()
+		}
+	}
+}
```

Comment:
instead of this, could we leverage https://pkg.go.dev/github.com/apache/arrow-go/v18/arrow/flight/flightsql#ColumnMetadata to handle the type management and the metadata values?